component/otelcol/processor/k8sattributes/k8sattributes.go

```diff
@@ -0,0 +1,257 @@
+	Labels    []FieldFilterConfig `river:"label,block,optional"`
+}
+
+func (args FilterConfig) convert() map[string]interface{} {
+	result := make(map[string]interface{})
+
+	if args.Node != "" {
+		result["node"] = args.Node
+	}
+
+	if args.Namespace != "" {
+		result["namespace"] = args.Namespace
+	}
+
+	fields := make([]interface{}, 0, len(args.Fields))
+	for _, field := range args.Fields {
+		fields = append(fields, field.convert())
+	}
+
+	if len(fields) > 0 {
+		result["fields"] = fields
+	}
+
+	labels := make([]interface{}, 0, len(args.Labels))
+	for _, label := range args.Labels {
+		labels = append(labels, label.convert())
+	}
+
+	if len(labels) > 0 {
+		result["labels"] = labels
+	}
+
+	return result
+}
+
+type PodAssociation struct {
+	Sources []PodAssociationSource `river:"source,block"`
+}
+
+func (args PodAssociation) convert() []map[string]interface{} {
+	result := make([]map[string]interface{}, 0, len(args.Sources))
+
+	for _, source := range args.Sources {
+		result = append(result, source.convert())
+	}
+
+	return result
+}
+
+type PodAssociationSource struct {
+	From string `river:"from,attr"`
+	Name string `river:"name,attr,optional"`
+}
+
+func (args PodAssociationSource) convert() map[string]interface{} {
+	return map[string]interface{}{
+		"from": args.From,
+		"name": args.Name,
+	}
+}
+
+type PodAssociationSlice []PodAssociation
+
+func (args PodAssociationSlice) convert() []map[string]interface{} {
+	result := make([]map[string]interface{}, 0, len(args))
+
+	for _, podAssociation := range args {
+		result = append(result, map[string]interface{}{
+			"sources": podAssociation.convert(),
+		})
+	}
+
+	return result
+}
+
+type ExcludeConfig struct {
+	Pods []ExcludePodConfig `river:"pod,block,optional"`
+}
+
+type ExcludePodConfig struct {
+	Name string `river:"name,attr"`
+}
+
+func (args ExcludePodConfig) convert() map[string]interface{} {
+	return map[string]interface{}{
+		"name": args.Name,
+	}
+}
+
+func (args ExcludeConfig) convert() map[string]interface{} {
+	result := make(map[string]interface{})
+
+	pods := make([]interface{}, 0, len(args.Pods))
+	for _, pod := range args.Pods {
+		pods = append(pods, pod.convert())
+	}
+	result["pods"] = pods
+
+	return result
+}
+
+// Arguments configures the otelcol.processor.k8sattributes component.
+type Arguments struct {
+	AuthType        string              `river:"auth_type,attr,optional"`
+	Passthrough     bool                `river:"passthrough,attr,optional"`
+	ExtractConfig   ExtractConfig       `river:"extract,block,optional"`
+	Filter          FilterConfig        `river:"filter,block,optional"`
+	PodAssociations PodAssociationSlice `river:"pod_association,block,optional"`
+	Exclude         ExcludeConfig       `river:"exclude,block,optional"`
+
+	// Output configures where to send processed data. Required.
+	Output *otelcol.ConsumerArguments `river:"output,block"`
+}
+
+var (
+	_ processor.Arguments = Arguments{}
```

Comment:
Consider adding a validate() function [like the one in the probabilistic sampler processor](https://github.com/grafana/agent/blob/main/component/otelcol/processor/probabilistic_sampler/probabilistic_sampler.go#L56).